sw/device/lib/testing/test_framework/ottf.h

```diff
@@ -32,6 +32,14 @@ typedef struct test_config {
    * by resetting the UART device before printing debug information.
    */
   bool can_clobber_uart;
+  /**
+   * If true, `test_main()` is run as a FreeRTOS task, enabling it to spawn
+   * additional (concurrent) FreeRTOS tasks if desired.
+   *
+   * If false, `test_main()` is executed on bare-metal. This is useful for tests
+   * that do not require concurrency, and seek to minimize simulation runtime.
+   */
+  bool run_as_freertos_task;
```

Comment:
Rather than enable concurrency at the test level, it may be useful to make it more fine grained - i.e. allow an individual test to spawn threads from within that actuate different pieces of hardware concurrently.

Reply:
Thats exactly what this enables. Once the `test_main()` is executing as a FreeRTOS task, test developers can call [xTaskCreate](https://www.freertos.org/a00125.html) with the `test_main()` (or any other function invoked in the main task) to create another "thread". Does that make sense?

Reply:
We could wrap FreeRTOS API calls too to make it easier for test writer to spawn a concurrent "thread" ("task" in FreeRTOS)

Reply:
I would second wrapping the API. Better than digging around in FreeRTOS's docs, and lets us customize to be specific to the testing use-case. For example, you might want the api to just be
```c
void ottf_spawn(const char* name, bool (*start)(void));
```
where it reports a test failure and halts the test if `start` returns `false` or something.

Reply:
SGTM! Tracking this as a separate task in #8015, and will submit a follow up PR with these wrappers to provide a clean OTTF API for test developers. Marking this as fixed.